src/package-request.js

```diff
@@ -136,6 +138,30 @@ export default class PackageRequest {
   }
 
   /**
+   * If pattern:
+   * - contains : it will return because is something like '<protocol>:<something else' or an url
+   * - is a file it will prepend file:
+   * otherwise it will return
+   */
+  async normalizeRange(pattern: string): Promise<string> {
+    if (pattern.includes(':')) {
+      return Promise.resolve(pattern);
+    }
+
+    if (await fs.exists(path.join(this.config.cwd, pattern))) {
+      return Promise.resolve(`file:${pattern}`);
+    }
+
+    return Promise.resolve(pattern);
+  }
+
+  async normalize(pattern: string): any {
+    const {name, range, hasVersion} = PackageRequest.normalizePattern(pattern);
+    const new_range = await this.normalizeRange(range);
```

Comment:
nit: do we use underscore for word separation anywhere in Yarn?
I think we should stick to the same camelCase.